solution.cpp

```diff
@@ -23,116 +23,143 @@ using std::ifstream;
+* `outfilename`, separated by newlines.
+*
+* Arguments:
+* - `size`: The size of (the number of elements in) the array.
+* - `primes`: An array of boolean values, with each element set to `true` if
+*   its index is prime, and `false` otherwise.
+* - `infilename`: A string representing the file path to read from.
+* - `outfilename`: A string representing the file path to write to.
+*
+* Returns:
+* - 0 on success, or -1 on error.
+*/
+int write_composites(const int size,
+	bool primes[],
+	const string& infilename,
+	const string& outfilename);
 
 // ----------------------------------------------------------------------------
 
 int main() {
-    int max = find_max("input.txt");
-    if (max == -1) {
-        cout << "ERROR in `find_max()`" << endl;
-        return 1;  // error
-    }
-
-    // - Runtime sized arrays are a C99 feature, and a GCC and Clang extension.
-    //   Thus `bool primes[max+1];` is not standards compliant.  It should work
-    //   on Xcode (which uses Clang) and in Linux (where most people use either
-    //   GCC or Clang), but Visual Studio appears not to allow it.
-    bool * primes = (bool *) malloc( sizeof(bool) * (max+1) );
-    sieve(max+1, primes);
-
-    int ret;  // for storing return values, to check for error codes
-    //
-    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_primes()`" << endl;
-        return 1;  // error
-    }
-    //
-    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_composites()`" << endl;
-        return 1;  // error
-    }
-
-    return 0;  // success
+	int max = find_max("input.txt");
+	if (max == -1) {
+		cout << "ERROR in `find_max()`" << endl;
+		return 1;  // error
+	}
+
+	// - Runtime sized arrays are a C99 feature, and a GCC and Clang extension.
+	//   Thus `bool primes[max+1];` is not standards compliant.  It should work
+	//   on Xcode (which uses Clang) and in Linux (where most people use either
+	//   GCC or Clang), but Visual Studio appears not to allow it.
+	bool * primes = (bool *)malloc(sizeof(bool)* (max + 1));
+	sieve(max + 1, primes);
+
+	int ret;  // for storing return values, to check for error codes
+	//
+	ret = write_primes(max + 1, primes, "input.txt", "primes.txt");
+	if (ret == -1) {
+		cout << "ERROR in `write_primes()`" << endl;
+		return 1;  // error
+	}
+	//
+	ret = write_composites(max + 1, primes, "input.txt", "composites.txt");
+	if (ret == -1) {
+		cout << "ERROR in `write_composites()`" << endl;
+		return 1;  // error
+	}
+
+	return 0;  // success
 }
 
 // ----------------------------------------------------------------------------
-
-// TODO: write prototyped functions
+int find_max(const string& infilename)
+
+{
+	int fields[50];
+	int index = 0;
+	int compare;
+	int top;
+
+	ifstream myfile1;
+	myfile1.open(infilename);
+
+	while (!myfile1.eof())
+	{
+		myfile1 >> fields[index];
+		index++
+	}
```

Comment:
this assumes that there will never be more than 50 integers in the input file, which is not a given assumption.  also, it's unnecessary to read them all in at the same time and store them, in order to find the largest one -- as i'm sure you've seen in my solution, by now :-)